go/login/server/example-server/cmd/run.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"../../../../glome"
+	"../../../server"
+
+	"github.com/spf13/cobra"
+)
+
+var (
+	keys        string
+	addr        string
+	script      string
+	userHeader  string
+	responseLen int
+
+	runCmd = &cobra.Command{
+		Use:   "run [-a <addr>] [-k <keyfile>] [-s <lua-script>]",
+		Short: "Run command creates a LoginServer that listen and serve in provided address.",
+		Long: `Run implements the server side of the glome-login protocol.
+Visit github.com/google/glome for more information on the protocol.
+
+This program implements the basic functionalities of the glome-server, using the glome-framework.
+It requires the flags -a, -k and -s to be provided to properly work. For more information, visit [add link to README.md]`,
+		Run: Run,
+	}
+)
+
+func Run(cmd *cobra.Command, args []string) {
+	if keys == "" || addr == "" || script == "" {
+		fmt.Println(`Please, provide key file, authorization script and address.
+For more information read --help option.`)
+		os.Exit(0)
+	}
```

Comment:
There's a mechanism for enforcing flag presence, https://github.com/spf13/cobra#required-flags. If you still need to do more complicated sanity checks (cross-deps between flags, for example), you can do that check in a `PreRunE`: https://github.com/spf13/cobra#prerun-and-postrun-hooks.

Reply:
I'd use a `PreRunE` that returns an `error` with that message. That would automatically terminate the process with a non-zero exit code.